### PR TITLE
feat: Add popup before hacker confirmation to collect location info

### PR DESCRIPTION
**Pull request by Tony9984**
### Tickets:

- HCK-#942

### List of changes:

- Added a confirm modal to prompt the hacker to enter location info (timeZone, country, city) before he/she confirms attendance
- Add enums for time zones and countries following the school enum structure
- Add additional props for the confirm modal to customize the cancel/confirm buttons 

### Type of change:

Please delete options that aren't relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [ ] Listed change(s) in the Changelog
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [x] Referenced all useful info (issues, tasks, etc)

### Screenshots:

After clicking the confirm button:

image

After filling in all info:

image

Updated modal styling:

image

Database update:

image

### CODE REVIEW

File: src/features/Status/StatusCTAContainer.tsx [hunk 1]
```diff
@@ -19,9 +26,16 @@ export interface IStatusCTAContainerState {
   status: HackerStatus;
   settings: ISetting;
   isModalOpen: boolean;
```
Comment (pierreTklein) [inline src/features/Status/StatusCTAContainer.tsx:21]: Can we specify which modal this is, now that we have two?

File: src/features/Status/StatusCTAContainer.tsx [hunk 2]
```diff
@@ -57,30 +78,94 @@ class StatusCTAContainer extends React.Component<IStatusCTAContainerProps, IStat
...
+            onClickConfirm={async () =>
+              this.setState({ isLocationModalOpen: true })
+            }
             // tslint:disable-next-line: jsx-no-lambda
             onClickWithdraw={() => this.setState({ isModalOpen: true })}
           />
         ) : (
           <ConfirmationEmailSentComponent />
         )}
+        <ConfirmModal
```
Comment (pierreTklein) [inline src/features/Status/StatusCTAContainer.tsx:116]: Can we refactor this component to another file?

File: src/features/Status/StatusCTAContainer.tsx [hunk 3]
```diff
@@ -57,30 +78,94 @@ class StatusCTAContainer extends React.Component<IStatusCTAContainerProps, IStat
...
+            <Label style={{ minWidth: '300px', maxWidth: '300px' }}>
+              In which city are you currently residing?
+              <Input
+                onChange={(e) => this.setState({ city: e.target.value })}
+                value={city}
+              />
+            </Label>
+          </Box>
+        </ConfirmModal>
         <ConfirmModal
```
Comment (pierreTklein) [inline src/features/Status/StatusCTAContainer.tsx:75]: What's the purpose of this modal?

File: src/shared/Elements/ConfirmModal.tsx [hunk 4]
```diff
@@ -38,7 +40,7 @@ export const ConfirmModal: React.FC<IConfirmModalProps> = (
               onClick={props.onConfirmed}
               variant={ButtonVariant.Primary}
             >
-              Yes
+              {props.confirmLabel || "Yes"}
```
Comment (pierreTklein) [inline src/shared/Elements/ConfirmModal.tsx:43]: Can "yes" and "no" be default values of the props?

File: src/features/Status/StatusCTAContainer.tsx (hunk 2 above)
Comment (Tony9984) [reply src/features/Status/StatusCTAContainer.tsx:117]: sure! should i put it in the status folder?

Comment (pierreTklein) [reply src/features/Status/StatusCTAContainer.tsx:117]: SGTM

File: src/features/Status/StatusCTAContainer.tsx (hunk 3 above)
Comment (Tony9984) [reply src/features/Status/StatusCTAContainer.tsx:75]: one is to collect info on the hacker's current location (which will probably be removed once we get off remote), and the other is a confirmation step before the hacker withdraws

File: src/shared/Elements/ConfirmModal.tsx (hunk 4 above)
Comment (Tony9984) [reply src/shared/Elements/ConfirmModal.tsx:43]: oops yeah i'll change it asap!

File: src/features/Status/StatusCTAContainer.tsx (hunk 3 above)
Comment (logan-r) [reply src/features/Status/StatusCTAContainer.tsx:75]: Yeah, these fields will be moved to the regular application form and only show up in remote hackathon mode. We didn't realize that we wanted to collect location+timezones info until after apps close so now are collecting them when they confirm

File: src/shared/Elements/ConfirmModal.tsx [hunk 5]
```diff
@@ -29,7 +31,7 @@ export const ConfirmModal: React.FC<IConfirmModalProps> = (
               onClick={props.onCanceled}
               variant={ButtonVariant.Primary}
```
Comment (logan-r) [inline src/shared/Elements/ConfirmModal.tsx:32]: Make this ButtonVariant.Secondary

File: src/features/Status/StatusCTAContainer.tsx [hunk 6]
```diff
@@ -57,30 +79,94 @@ class StatusCTAContainer extends React.Component<IStatusCTAContainerProps, IStat
...
           />
         ) : (
           <ConfirmationEmailSentComponent />
         )}
+        <ConfirmModal
+          isOpen={this.state.isLocationModalOpen}
+          onCanceled={() => this.setState({ isLocationModalOpen: false })}
+          onConfirmed={this.confirmStatus}
+          cancelLabel="Cancel"
+          confirmLabel="Submit"
```
Comment (logan-r) [inline src/features/Status/StatusCTAContainer.tsx:122]: Change confirmLabel="Submit" to confirmLabel="Confirm"

File: src/features/Status/StatusCTAContainer.tsx [hunk 7]
```diff
@@ -57,30 +79,94 @@ class StatusCTAContainer extends React.Component<IStatusCTAContainerProps, IStat
...
+        >
+          <Box mb={'10px'} alignSelf={'center'}>
```
Comment (logan-r) [inline src/features/Status/StatusCTAContainer.tsx:124]: Can you add a title of "Confirm you spot" and a subtitle of "Please let us know where you'll be hacking from"? (or some text along those lines)

**Review by logan-r — CHANGES_REQUESTED** [review]

File: src/features/Status/StatusCTAContainer.tsx [hunk 8]
```diff
@@ -19,9 +21,16 @@ export interface IStatusCTAContainerState {
   status: HackerStatus;
   settings: ISetting;
   isModalOpen: boolean;
```
Comment (logan-r) [inline src/features/Status/StatusCTAContainer.tsx:23]: nit: rename `isWithdrawModalOpen`

**Review by logan-r — CHANGES_REQUESTED** [review]
One small nit, otherwise looks good to merge

File: src/features/Status/StatusCTAContainer.tsx [hunk 9]
```diff
@@ -57,30 +74,58 @@ class StatusCTAContainer extends React.Component<IStatusCTAContainerProps, IStat
...
+        />
         <ConfirmModal
-          isOpen={this.state.isModalOpen}
+          isOpen={this.state.isWithdrawModalOpen}
           // tslint:disable-next-line: jsx-no-lambda
-          onCanceled={() => this.setState({ isModalOpen: false })}
+          onCanceled={() => this.setState({ isWithdrawModalOpen: false })}
           // tslint:disable-next-line: jsx-no-lambda
-          onConfirmed={() => {
-            this.withdrawStatus();
```
Comment (pierreTklein) [inline src/features/Status/StatusCTAContainer.tsx:81]: Why was the withdrawStatus function removed?

File: src/features/Status/StatusCTAContainer.tsx [hunk 10]
```diff
@@ -57,30 +74,58 @@ class StatusCTAContainer extends React.Component<IStatusCTAContainerProps, IStat
...
+        <StatusCTALocationModal
+          isModalOpen={this.state.isLocationModalOpen}
+          onCanceled={() => this.setState({ isLocationModalOpen: false })}
+          onConfirmed={this.confirmStatus}
+          timeZone={timeZone}
+          country={country}
+          city={city}
+          handleChangeTimeZone={this.handleChangeTimeZone}
+          handleChangeCountry={this.handleChangeCountry}
+          handleChangeCity={console.log('test')}
```
Comment (pierreTklein) [inline src/features/Status/StatusCTAContainer.tsx:121]: Please fix!

**Review by pierreTklein — CHANGES_REQUESTED** [review]

File: src/features/Status/StatusCTAContainer.tsx (hunk 9 above)
Comment (Tony9984) [reply src/features/Status/StatusCTAContainer.tsx:81]: error on my part 😓

**Review by logan-r — APPROVED** [review]

**Comment by logan-r** [comment]
Merging now since need to release applications decision asap

**Pull request merged**